File: leetcode/Python/last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        pq = list()
        for s in stones:
            heapq.heappush(pq, -s)
        while len(pq) > 1:
            a = -heapq.heappop(pq)
            b = -heapq.heappop(pq)
            if a>b:
                heapq.heappush(pq, b-a)
        return 0 if len(pq)==0 else -pq[0]

File: leetcode/C++/last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq;
        int a=0,b=0;
        for(int &s : stones) pq.push(s);
        while(pq.size() > 1){
            a=pq.top(); pq.pop(); b=pq.top(); pq.pop();
            if(a>b) pq.push(a-b);
        }
        return (pq.empty())? 0 : pq.top();
    }
};